rclcpp/include/rclcpp/detail/resolve_use_intra_process.hpp

```diff
@@ -26,12 +26,12 @@ namespace detail
 {
 
 /// Return whether or not intra process is enabled, resolving "NodeDefault" if needed.
-template<typename OptionsT, typename NodeBaseT>
+template<typename NodeBaseT>
 bool
-resolve_use_intra_process(const OptionsT & options, const NodeBaseT & node_base)
+resolve_use_intra_process(const IntraProcessSetting & ipc_setting, const NodeBaseT & node_base)
```

Comment:
`IntraProcessSetting` is an enum, we can avoid passing that as a reference

Reply:
Done in [d19d5c2](https://github.com/ros2/rclcpp/pull/1847/commits/d19d5c21efa9c70592186747d2f0f0aa572069b8)